### PR TITLE
refactor(log): use LOG_WARNING_F instead of LOG_WARNING (1/3)

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
#1305

This patch aim to use LOG_WARNING_F instead of LOG_WARNING for replica module. Includes:
- Use LOG_WARNING_F instead of LOG_WARNING in replica module.
- Use LOG_WARNING_PREFIX instaed of LOG_WARNING for class replica

### CODE REVIEW

File: src/replica/mutation_log.cpp [hunk 1]
```diff
@@ -997,26 +996,26 @@ int64_t mutation_log::on_partition_reset(gpid gpid, decree max_decree)
...
-                    old_info.valid_start_offset,
-                    _private_log_info.valid_start_offset);
+        LOG_WARNING_F("replica {}.{} has changed private log max_decree from {} to {}, "
+                      "valid_start_offset from {} to {}",
+                      gpid,
+                      gpid.get_partition_index(),
+                      old_info.max_decree,
+                      _private_log_info.max_decree,
+                      old_info.valid_start_offset,
+                      _private_log_info.valid_start_offset);
```
Comment (acelyc111) [inline src/replica/mutation_log.cpp:1006]:
```suggestion
        LOG_WARNING_F("replica {} has changed private log max_decree from {} to {}, "
                      "valid_start_offset from {} to {}",
                      gpid,
                      old_info.max_decree,
                      _private_log_info.max_decree,
                      old_info.valid_start_offset,
                      _private_log_info.valid_start_offset);
```

File: src/replica/mutation_log.cpp [hunk 2]
```diff
@@ -997,26 +996,26 @@ int64_t mutation_log::on_partition_reset(gpid gpid, decree max_decree)
...
-                        it.first->second.valid_start_offset,
-                        info.valid_start_offset);
+            LOG_WARNING_F("replica {}.{} has changed shared log max_decree from {} to {}, "
+                          "valid_start_offset from {} to {} ",
+                          gpid,
+                          gpid.get_partition_index(),
+                          it.first->second.max_decree,
+                          info.max_decree,
+                          it.first->second.valid_start_offset,
+                          info.valid_start_offset);
```
Comment (acelyc111) [inline src/replica/mutation_log.cpp:1018]:
```suggestion
            LOG_WARNING_F("replica {} has changed shared log max_decree from {} to {}, "
                          "valid_start_offset from {} to {} ",
                          gpid,
                          it.first->second.max_decree,
                          info.max_decree,
                          it.first->second.valid_start_offset,
                          info.valid_start_offset);
```

File: src/replica/mutation_log.cpp (hunk 1 above)
Comment (WHBANG) [reply src/replica/mutation_log.cpp]: done

File: src/replica/mutation_log.cpp (hunk 2 above)
Comment (WHBANG) [reply src/replica/mutation_log.cpp]: done

File: src/replica/replica_check.cpp [hunk 3]
```diff
@@ -160,12 +158,12 @@ void replica::on_group_check(const group_check_request &request,
 
     if (request.config.ballot < get_ballot()) {
         response.err = ERR_VERSION_OUTDATED;
-        LOG_WARNING("%s: on_group_check reply %s", name(), response.err.to_string());
+        LOG_WARNING_PREFIX("on_group_check reply {}", response.err);
```
[the remark this answers is not in the document]
Comment (WHBANG) [reply src/replica/replica_check.cpp:161]: Thank you for your valuable comments

_ PREFIX can print  _name of the class. class replica_check inherits class replica, class replica inherits replica_base, and replica_base implements the method of print _name, so there is no problem

File: src/replica/replica_learn.cpp [hunk 4]
```diff
@@ -178,15 +173,15 @@ void replica::init_learn(uint64_t signature)
...
-                    _stub->_learn_app_concurrent_count.load(),
-                    _options->learn_app_max_concurrent_count);
+        LOG_WARNING_PREFIX(
+            "init_learn[{:#018x}]: learnee = {}, learn_duration = {} ms, need to learn app "
+            "because app_committed_decree = 0, but learn_app_concurrent_count({}) >= "
+            "learn_app_max_concurrent_count({}), skip",
+            _potential_secondary_states.learning_version,
+            _config.primary,
+            _potential_secondary_states.duration_ms(),
+            _stub->_learn_app_concurrent_count.load(),
```
Comment (empiredan) [inline src/replica/replica_learn.cpp:183]:
```suggestion
            _stub->_learn_app_concurrent_count,
```

File: src/replica/replica_learn.cpp [hunk 5]
```diff
@@ -694,14 +685,13 @@ void replica::on_learn_reply(error_code err, learn_request &&req, learn_response
...
-                        _potential_secondary_states.learning_version,
-                        _config.primary.to_string(),
-                        _stub->_learn_app_concurrent_count.load(),
-                        _options->learn_app_max_concurrent_count);
+            LOG_WARNING_PREFIX(
+                "on_learn_reply[{:#018x}]: learnee = {}, learn_app_concurrent_count({}) >= "
+                "learn_app_max_concurrent_count({}), skip this round",
+                _potential_secondary_states.learning_version,
+                _config.primary,
+                _stub->_learn_app_concurrent_count.load(),
```
Comment (empiredan) [inline src/replica/replica_learn.cpp:693]:
```suggestion
                _stub->_learn_app_concurrent_count,
```

**Review by empiredan — APPROVED** [review]

**Pull request merged**